pkg/vdr/key/creator.go

```diff
@@ -88,10 +97,82 @@ func (v *VDR) Create(keyManager kms.KeyManager, didDoc *did.Doc,
 	return &did.DocResolution{DIDDocument: createDoc(publicKey, keyAgr, didKey)}, nil
 }
 
+func getKeyTypeAndVerMethodType(kt kms.KeyType) (kms.KeyType, string) {
+	var (
+		keyType kms.KeyType
+		vmType  string
+	)
+
+	switch kt {
+	case kms.ED25519Type:
+		keyType = kms.ED25519Type
+		vmType = ed25519VerificationKey2018
+	case kms.ECDSAP256TypeDER:
+		keyType = kms.ECDSAP256TypeDER
+		vmType = jsonWebKey2020
+	case kms.ECDSAP256TypeIEEEP1363:
+		keyType = kms.ECDSAP256TypeIEEEP1363
+		vmType = jsonWebKey2020
+	case kms.ECDSAP384TypeDER:
+		keyType = kms.ECDSAP384TypeDER
+		vmType = jsonWebKey2020
+	case kms.ECDSAP384TypeIEEEP1363:
+		keyType = kms.ECDSAP384TypeIEEEP1363
+		vmType = jsonWebKey2020
+	case kms.ECDSAP521TypeDER:
+		keyType = kms.ECDSAP521TypeDER
+		vmType = jsonWebKey2020
+	case kms.ECDSAP521TypeIEEEP1363:
+		keyType = kms.ECDSAP521TypeIEEEP1363
+		vmType = jsonWebKey2020
+	case kms.BLS12381G2Type:
+		keyType = kms.BLS12381G2Type
+		vmType = bls12381G2Key2020
+	default:
+		keyType = kms.ED25519Type
+		vmType = ed25519VerificationKey2018
+	}
+
+	return keyType, vmType
+}
+
+func getKeyCode(keyType kms.KeyType, verificationMethod *did.VerificationMethod) (uint64, error) {
+	var keyCode uint64
+
+	switch verificationMethod.Type {
+	case ed25519VerificationKey2018:
+		keyCode = ed25519Pub
+	case bls12381G2Key2020:
+		keyCode = bls12381g2Pub
+	case jsonWebKey2020:
+		switch keyType {
```

Comment:
For more cleaner code, the `keyCode` should be derived from the values of the verificationMethod.

Reply:
yes, `keyCode` is derived from the values of verificationMethod, but if keyType option was used (ie creating a new kms key and VerificatonMethod entry) then keyType is more specific. See my previous comment [above](https://github.com/hyperledger/aries-framework-go/pull/2594#discussion_r587078042)